seqr/views/utils/terra_api_utils.py

```diff
@@ -162,6 +162,12 @@ def list_anvil_workspaces(user):
     return r
 
 
+def user_get_workspace_bucket(user, workspace_namespace, workspace_name):
+    path = "api/workspaces/{0}/{1}?fields=workspace.bucketName".format(workspace_namespace, workspace_name)
+    r = _user_anvil_call('get', path, user)
+    return r['workspace']['bucketName']
```

Comment:
The unit test for this function is to be added.

Reply:
So I though I have is that when you call `check_workspace_perm`, that ends up calling `user_get_workspace_access_level` which calls this exact endpoint but with a different `fields` query param. What if `check_workspace_perm` and `has_workspace_perm` has a keyword arg called `get_workspace_meta` and if it was True both of those functions would pass an extra `fields` argument to `user_get_workspace_access_level` to return extra data from that request and then we would only need to query the bucket once. So something like this:
```
def check_workspace_perm(user, permission_level, namespace, name, can_share=False, get_workspace_meta=False):
    workspace_perm = has_workspace_perm(user, permission_level, namespace, name, can_share, get_workspace_meta)
    if workspace_perm:
        return workspace_perm
   # rest of the function stays the same

...

def has_workspace_perm(user, permission_level, namespace, name, can_share=False, get_workspace_meta=False):
    workspace_permission = user_get_workspace_access_level(user, namespace, name, fields=['workspace.bucketName'] if get_workspace_meta else None)
    ...
    if not has_permission:
       return False
    return workspace_permission['workspace'] if get_workspace_meta else True

...

def user_get_workspace_access_level(user, workspace_namespace, workspace_name, fields=None):
    query_fields = ['accessLevel', 'canShare'] + (fields or []) 
    path = "api/workspaces/{0}/{1}?fields={2}".format(workspace_namespace, workspace_name, ','.join(query_fields))
    # rest of the function stays the same

```

Reply:
Sound good!